### ISSUE TITLE
Label format for Training Custom Dataset?

### ISSUE THREAD

**Issue by llavkush** [ultralytics/ultralytics#1259]
I have labelled dataset in COCO JSON format for Instance Segmentation Project, Please let me know how to convert it into for training using Yolo8 Instance segmentation or what is the format for yolo8 txt file annotation, I have tried using roboflow for converting but its not able to recognize the labels, P:S I have used same coco format for training on detectron2 successfully. need to shift to yolo for inference on edge device

**Comment by AyushExel** [comment]
You can use this repo to convert labels from json to yolo format https://github.com/ultralytics/JSON2YOLO

**Comment by llavkush** [comment]
Does it convert labels to use it for instance segmentation projects?, last time I checked it was only able  to export bounding box annotations not pixel-level annotations

**Comment by alevangel** [comment]
How to use instead a custom dataloader?

**Comment by ryouchinsa** [comment]
If you are using RLE masks in your COCO JSON format, you can use this script to convert to the YOLOv5/v8 segmentation format.
https://github.com/ultralytics/JSON2YOLO/issues/38

If you could show us your COCO JSON format here, we can check the script works or not.

**Comment by pderrenger** [comment]
@ryouchinsa if you are looking to use a custom dataloader, you can create a custom dataset class in Python using PyTorch or another deep learning framework. This class will read your COCO JSON format and convert the label format to YOLOv8 segmentation format on-the-fly. You can then use this custom dataloader to load your dataset for training and validation.

You can reference the Ultralytics documentation for creating custom dataset classes in PyTorch to help you get started. If you encounter any specific issues while implementing your custom dataloader, feel free to share the details here, and we can provide further assistance.
